src/librustc_target/abi/mod.rs

```diff
@@ -619,13 +619,16 @@ impl Scalar {
 /// Describes how the fields of a type are located in memory.
 #[derive(PartialEq, Eq, Hash, Debug, HashStable_Generic)]
 pub enum FieldsShape {
+    Primitive,
```

Comment:
Please add a doc comment.

Reply:
Hmm maybe this?
```rust
/// Scalar primitives and `!`, which never have fields.
```

Alternatively, "uninhabited ZSTs" (until we fix `enum`s) instead of just `!`.